src/metabase/server/middleware/session.clj

```diff
@@ -359,21 +359,46 @@
 ;;; |                                              reset-cookie-timeout                                             |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn- check-session-timeout
+  "Returns nil if the [[session-timeout]] value is valid. Otherwise returns an exception."
+  [timeout]
+  (when (some? timeout)
+    (let [{:keys [unit amount]} timeout
+          units-in-24-hours (case unit
+                              "seconds" (* 60 60 24)
+                              "minutes" (* 60 24)
+                              "hours"   24)
+          units-in-100-years (* units-in-24-hours 365.25 100)]
+      (cond
+        (not (pos? amount))
+        (Exception. (tru "Session timeout amount must be positive."))
```

Comment:
Maybe instead of returning Exceptions directly this should just return some sort of error keyword, that way you can i18n it correctly based on whether it's logged or thrown as an Exception